indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchSupervisorStatus.java

```diff
@@ -21,97 +21,137 @@
+  private final Integer totalSubmittedTasks;
 
-  @JsonProperty
-  private final Map<String, TaskStatus> completedTasks;
+  private final Integer totalSuccessfulTasks;
+
+  private final Integer totalFailedTasks;
+
+  private final List<BatchSupervisorTaskStatus> recentActiveTasks;
+
+  private final List<BatchSupervisorTaskStatus> recentSuccessfulTasks;
+
+  private final List<BatchSupervisorTaskStatus> recentFailedTasks;
 
   @JsonCreator
-  public ScheduledBatchSupervisorSnapshot(
+  public ScheduledBatchSupervisorStatus(
       @JsonProperty("supervisorId") String supervisorId,
-      @JsonProperty("status") BatchSupervisorStatus status,
+      @JsonProperty("status") ScheduledBatchSupervisor.State state,
       @JsonProperty("lastTaskSubmittedTime") @Nullable DateTime lastTaskSubmittedTime,
       @JsonProperty("nextTaskSubmissionTime") @Nullable DateTime nextTaskSubmissionTime,
       @JsonProperty("timeUntilNextTaskSubmission") @Nullable Duration timeUntilNextTaskSubmission,
-      @JsonProperty("activeTasks") Map<String, TaskStatus> activeTasks,
-      @JsonProperty("completedTasks") Map<String, TaskStatus> completedTasks
+      @JsonProperty("totalSubmittedTasks") Integer totalSubmittedTasks,
+      @JsonProperty("totalSuccessfulTasks") Integer totalSuccessfulTasks,
+      @JsonProperty("totalFailedTasks") Integer totalFailedTasks,
+      @JsonProperty("recentActiveTasks") List<BatchSupervisorTaskStatus> recentActiveTasks,
+      @JsonProperty("recentSuccessfulTasks") List<BatchSupervisorTaskStatus> recentSuccessfulTasks,
+      @JsonProperty("recentFailedTasks") List<BatchSupervisorTaskStatus> recentFailedTasks
   )
   {
     this.supervisorId = supervisorId;
-    this.status = status;
+    this.state = state;
     this.lastTaskSubmittedTime = lastTaskSubmittedTime;
     this.nextTaskSubmissionTime = nextTaskSubmissionTime;
     this.timeUntilNextTaskSubmission = timeUntilNextTaskSubmission;
-    this.activeTasks = activeTasks;
-    this.completedTasks = completedTasks;
+    this.totalSubmittedTasks = totalSubmittedTasks;
+    this.totalSuccessfulTasks = totalSuccessfulTasks;
+    this.totalFailedTasks = totalFailedTasks;
+    this.recentActiveTasks = recentActiveTasks;
+    this.recentSuccessfulTasks = recentSuccessfulTasks;
+    this.recentFailedTasks = recentFailedTasks;
   }
 
+  @JsonProperty
   public String getSupervisorId()
   {
     return supervisorId;
   }
 
-  public ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus getStatus()
+  @JsonProperty
+  public ScheduledBatchSupervisor.State getState()
   {
-    return status;
+    return state;
   }
 
+  @Nullable
+  @JsonProperty
   public DateTime getLastTaskSubmittedTime()
   {
     return lastTaskSubmittedTime;
   }
 
+  @Nullable
+  @JsonProperty
   public DateTime getNextTaskSubmissionTime()
   {
     return nextTaskSubmissionTime;
   }
 
+  @Nullable
+  @JsonProperty
   public Duration getTimeUntilNextTaskSubmission()
```

Comment:
Is serializing this field necessary? I think `getNextTaskSubmissionTime` should be sufficient.

Reply:
I've removed this for now